### ISSUE TITLE
Picture downloading not progressively after setting to download progressively

### ISSUE THREAD

**Issue by Johnkui** [pinterest/PINRemoteImage#168]
In the published Example(ProgressiveViewController) I changed the picture address(https://s-media-cache-ak0.pinimg.com/1200x/2e/0c/c5/2e0cc5d86e7b7cd42af225c29f21c37f.jpg) to another one(http://38ccda.medialib.glogster.com/media/7c744f06bf559a9be5fcb84f6b9e6f887e92ed0f74fe4b2e4192727fb0f943f1/grassland-summer.jpg) to download it progressively, but the it turned out not loading picture progressively. So I am wondering why? any format requirement for the downloaded picture?

**Comment by garrettmoon** [comment]
Hi @Johnkui there are two conditions that need to be met for an jpeg to be rendered progressively.

1. The image must be encoded progressively. As you mention, this is a different format. You can learn about progressive encoding of jpegs pretty easily by Googling.
2. PINRemoteImage will only render a progressive scan of an image if it thinks that it will take a little while longer to download the image – essentially, it doesn't waste the CPU time to render the progressive image if the rest of the image will be done downloading momentarily. This is why the sample project suggests you turn on the Network Link Conditioner to simulate a slow internet connection.

In summary, the image needs to be progressively encoded and you need to be on a slow internet connection. Hope this helps!

**Comment by Johnkui** [comment]
Thank you, @garrettmoon!  So PINRemoteImage does not support progressively downloading for the baseline and interlace encoded image, right?

**Comment by garrettmoon** [comment]
I'm no expert by any means, but interlacing is not supported by jpeg though progressive is sometimes referred to as interlacing.

You're correct that progressive rendering is *not* supported on baseline images, but it wouldn't betook difficult to add support for it.
